server/middlewares/Authentication.js

```diff
@@ -0,0 +1,227 @@
+    if (!request.body.password || !request.body.email) {
+      return response.status(400)
+        .send({
+          message: 'Please provide your email and password to login.'
+        });
+    }
+    const email = /\S+@\S+\.\S+/.test(request.body.email),
+      password = /\w+/g.test(request.body.password);
+    if (!email || !password) {
+      return response.status(400)
+        .send({
+          message: 'Please enter a valid email and password.'
+        });
+    }
+    next();
+  },
+  /**
+   * Checks users' input
+   * @param  {Object} request  Request object
+   * @param  {Object} response  Response object
+   * @param  {Object} next Next process handler
+   * @return {Object}
+   */
+  verifyUserInput(request, response, next) {
+    let email = /\S+@\S+\.\S+/.test(request.body.email),
+      firstName = /\w+/g.test(request.body.firstname),
+      lastName = /\w+/g.test(request.body.lastname),
+      userName = /\w+/g.test(request.body.username),
+      password = /\w+/g.test(request.body.password);
+
+    if (!email) {
+      return response.status(400)
+        .send({
+          message: 'Please enter a valid email address.'
+        });
+    }
+    if (!firstName || firstName === 'undefined') {
+      return response.status(400)
+        .send({
+          message: 'Please enter a valid firstname.'
+        });
+    }
+    if (!lastName || lastName === 'undefined') {
+      return response.status(400)
+        .send({
+          message: 'Please enter a valid lastname.'
+        });
+    }
+    if (!userName || userName === 'undefined') {
+      return response.status(400)
+        .send({
+          message: 'Please enter a valid username.'
+        });
+    }
+    if (!password || password === 'undefined') {
+      return response.status(400)
+        .send({
+          message: 'Password field cannot be empty. Please enter a password.'
+        });
+    }
+    if (request.body.password.length < 8) {
+      return response.status(400)
+        .send({
+          message: 'Password cannot be less than 8 characters. Try again.'
+        });
+    }
+    db.User.findOne({ where: { email: request.body.email } })
+      .then((user) => {
+        if (user) {
+          return response.status(409)
+            .send({
+              message: 'Email address already exist.'
+            });
+        }
+        db.User.findOne({ where: { username: request.body.username } })
+          .then((newUser) => {
+            if (newUser) {
+              return response.status(409)
+                .send({
+                  message: 'Username already exist.'
+                });
+            }
+            userName = request.body.username;
+            firstName = request.body.firstname;
+            lastName = request.body.lastname;
+            email = request.body.email;
+            password = request.body.password;
+
+            const roleId = request.body.roleId || 2;
+            request.userInput = {
```

Comment:
why are you rewriting the values to userInput, you can just add additional value to req.body?